### ISSUE TITLE
LDAP Problems

### ISSUE THREAD

**Issue by inDane** [openmicroscopy/ome-documentation#2005]
I've had problems with the LDAP settings. I solved my problems, but I'd like to share my experience. You can perhaps add some lines to the documentation, so that other people don't need to make the same mistakes.

-----

The first problem was that i have some special characters in my Password:
`OMERO.server/bin/omero config set omero.ldap.password "hel!o" `
results in an error. Then i thought, i can just use \ to except that. 
`OMERO.server/bin/omero config set omero.ldap.password "hel\!o"`
But this does actually get used for authentication, which resulsts in a bad credentials error. You can see the config in  `/OMERO.server/etc/grid/config.xml`

This is the error stated in `log/Blitz-0.log`
> [LDAP: error code 49 - 80090308: LdapErr: DSID-0C09042A, comment: AcceptSecurityContext error, data 52e, v3839]; nested exception is javax.naming.AuthenticationException: [LDAP: error code 49 - 80090308: LdapErr: DSID-0C09042A, comment: AcceptSecurityContext error, data 52e, v3839]

-----

The next problem that occurred was 

> ome.conditions.InternalException:  Wrapped Exception: (org.springframework.ldap.PartialResultException):
Unprocessed Continuation Reference(s); nested exception is javax.naming.PartialResultException: Unprocessed Continuation Reference(s); remaining name ''

This is, because omero.ldap.referral default is false. For our Active Directory we need omero.ldap.referral follow
`OMERO.server/bin/omero config set omero.ldap.referral "follow"`

-----

> ome.conditions.ValidationException: not-null property references a null or transient value: ome.model.meta.Experimenter.lastName; nested exception is org.hibernate.PropertyValueException: not-null property references a null or transient value: ome.model.meta.Experimenter.lastName

The last problem getting LDAP to work was the user_mapping. I've customized the option and cut some parts of it, which i thought were unnecessary.  Every variable needs to be filled! Otherwise it will fail with the error stated above.

`OMERO.server/bin/omero config set omero.ldap.user_mapping "omeName=sAMAccountName, firstName=givenName, lastName=sn, email=mail, institution=department, middleName=middleName"`

I hope this is the right place to put this inquiry.

**Comment by mtbc** [comment]
Thank you for so clearly and specifically describing the issues you ran into, this will certainly help us to improve the documentation and/or error reporting for future users; we generally aim for a trend of OMERO becoming *easier* to install! This is a fine place to open the issue; I'll leave it open to ensure it stays on our radar.

**Comment by glyg** [comment]
Hi,

Writing here as the issue is still opened, 

I ran in the same kind of problem, and setting all the variables in the mapping was indeed a solution, but I have still an issue with the ldap authentication because the `givenName` attribute is not present for all the users in the Active Directory server I'm querying. I'm not the administrator of that directory, and would prefer to keep a read only role.

Is it possible to directly map the whole `cn` to the firstName, lastName entries in OME?

I would  also be fine for me to simply catch the exception and rise a warning?

**Comment by joshmoore** [comment]
Hi @glyg,

I think this may be a separate issue, but trying to handle it here before we do anything more heavy weight:

> Is it possible to directly map the whole cn to the firstName, lastName entries in OME?


Have you tried to list it twice? `firstName=cn, lastName=cn` -- it's not going to be overly attractive but it might unblock you. Otherwise, are there any other useful fields? (In OMERO, first and last name are required so a warning won't be possible)

~J.

**Comment by glyg** [comment]
HI @joshmoore 
The sys admins kindly fixed the database, so the problem kind of got away but  I'm sure your fix would have worked
Thanks, I'll close the other issue with a ref to here.
